src/GNDStk/BodyText.hpp

```diff
@@ -57,56 +42,74 @@ class BodyText {
 
 template<>
 class BodyText<true> {
+public:
 
-   // Primary data:
-   // Raw string, and variant of vectors
-   #include "GNDStk/BodyText/src/data.hpp"
+   #include "GNDStk/BodyText/src/types.hpp"
 
-   // Parameters that affect interpretation of the raw string:
-   // vars { length, start, and valueType }
-   #include "GNDStk/BodyText/src/params.hpp"
+private:
+
+   // For internal use
+   enum class Active { string, vector };
+   mutable Active active = Active::string;
+
+   // Raw string, directly from "plain character data" in a GNDS file.
+   // We'll allow callers to set this by using a setter.
+   std::string rawstring;
+
+   // Vector of <several possibilities>.
+   // Mutable, so that we can defer processing of the raw string into
+   // a vector until, and unless, a caller *asks* for the vector.
+   mutable VariantOfVectors variant;
 
 public:
 
-   enum class Active {
-      string,
-      vector
-   };
-   mutable Active active = Active::string;
+   // Parameters that affect interpretation of the raw string:
+   //    struct vars { length, start, valueType }
+   // Includes public getters and setters for those.
+   #include "GNDStk/BodyText/src/params.hpp"
 
    // trim
-   // This indicates whether the process of converting a BodyText object back
-   // into data - in a Node - should, or should not, trim zeros from the start
-   // and/or the end of the vector.
+   // Flag: should the conversion of BodyText data back into textual data, in
+   // a Node, trim zeros from the start and end of the output?
    mutable bool trim = true;
 
-   // Getters and setters for length, start, and valueType:
-   #include "GNDStk/BodyText/src/setters.hpp"
-
    // Getters and setters for the raw string:
    #include "GNDStk/BodyText/src/string.hpp"
 
-   // Simple member functions:
-   // clear() and size()
-   #include "GNDStk/BodyText/src/functions.hpp"
-
-   // Various get() functions, including type-specific ones (doubles() etc.)
+   // clear()
+   // Clears the active vector alternative in the variant.
+   BodyText &clear()
+   {
+      std::visit([](auto &&alt) { alt.clear(); }, variant);
+      return *this;
```

Comment:
Maybe, but I'm not sure about this. For now, I'm going to presume that someone who clears the vector still wants the vector (not the string) to be active; they're just preparing to fill it with new values. If someone wants a string instead, just use the assignment from string. (Or the string() setter.)

Reply:
Wait, I guess you're saying that if the string is currently active, then someone calls clear(), then they may intend for a vector to become active. Otherwise they wouldn't be calling something (clear()) that deals with the vector.

Reply:
Will be changed in next PR.